ecl2df/gruptree.py

```diff
@@ -71,38 +71,40 @@ def df(
     if isinstance(deck, EclFiles):
         deck = deck.get_ecldeck()
 
-    gruptreerecords = []  # list of dict of rows containing an edge.
-    grupnetrecords = []
+    edgerecords = []  # list of dict of rows containing an edge.
+    nodedatarecords = []
 
-    # In order for the GRUPTREE keywords to accumulate, we
+    # In order for the GRUPTREE/BRANPROP keywords to accumulate, we
     # store the edges as a dictionary indexed by the edge
     # (which is a tuple of child and parent).
     # The value of the dictionary is GRUPTREE or WELSPECS
-    currentedges: Dict[tuple, str] = dict()
-
-    grupnet_df: pd.DataFrame = pd.DataFrame()
-
-    found_gruptree = False  # Flags which will tell when a new GRUPTREE or
-    found_welspecs = False  # WELSPECS have been encountered.
-    found_grupnet = False  # GRUPNET has been encountered
+    currentedges: Dict[str, Dict[tuple, str]] = {"GRUPTREE": dict(), "BRANPROP": dict()}
+    wellspecsedges: Dict[tuple, str] = dict()
+    nodedata: Dict[str, pd.DataFrame] = {
+        "GRUPNET": pd.DataFrame(),
+        "NODEPROP": pd.DataFrame(),
+    }
+
+    # Flags which will tell when a new GRUPTREE/BRANPROP, WELSPECS or
+    # GRUPNET/NODEPROP have been encountered
+    keywords = ["GRUPTREE", "BRANPROP", "WELSPECS", "GRUPNET", "NODEPROP"]
+    found_keywords = {key: False for key in keywords}
     for kword in deck:
         if kword.name == "DATES" or kword.name == "START" or kword.name == "TSTEP":
             # Whenever we encounter a new DATES, it means that
             # we have processed all the GRUPTREE and WELSPECS that
             # have occured since the last date, so this is the chance
             # to dump the parsed data. Also we dump the *entire* tree
             # at every date with a change, not only the newfound edges.
-            if currentedges and (found_gruptree or found_welspecs or found_grupnet):
+            if any([val for val in found_keywords.values()]):
```

Comment:
check if pylint is complaining on this line, I wonder if the square brackets can be removed.

Reply:
I will check

Reply:
pylint didn't complain, but it works without the brackets so I'll take them away.